sources/engine/Stride.Rendering/Rendering/Mesh.cs

```diff
@@ -76,5 +82,165 @@ public Mesh(MeshDraw meshDraw, ParameterCollection parameters)
+                        MATBSHAPE[iBendShape * Draw.VertexMapping.Length + iVert].Row1 = new Vector4(vectorValue, blendShapesWeights[iBendShape]);
+
+                    }
+                }
+                BlendShapeProcessingNecessary = false;
+            }
+        }
+        
+
+        public void AddBlendShapes(Shape shape, float weight)
+        {
+            (Shapes ??= new()).Add(shape, weight); 
+            Parameters.Set(MaterialKeys.HasBlendShape, true);
+            
+        }
+
+        public float[] GetBlendWeights(out float cummulativeWeight)
+        {
+            float[] shapeWeights = new float[Shapes.Count];
+            cummulativeWeight = 0f;
+             
+            for (int i = 0; i < Shapes.Count; i++)
+            {
+                var shapeKV = Shapes.ElementAt(i);
+                var shape = shapeKV.Key;
+                var shapeWeight = Math.Clamp(shapeKV.Value, 0f, 1f);
+                float adjustedWeight = shapeWeight;
+               // if (cummulativeWeight >= 1f) { adjustedWeight = 0f; }
+               // else if (cummulativeWeight + shapeWeight > 1) { adjustedWeight = 1 - cummulativeWeight; }
+                //else
+                //{
+                  //  adjustedWeight = shapeWeight;
+                //}
+                cummulativeWeight += adjustedWeight;
+                shapeWeights[i] = adjustedWeight;
+            }
+            return shapeWeights;
+        }
+
+
+  
+        public List<Vector3> AdjustPositionToDrawInstance()
+        {
+            List<Vector3> adjustedPositons = new List<Vector3>();
+            foreach (var shape in Shapes)
+            {
+                var posBlend = shape.Key.Position;
+                List<int> originalVerticesIDS = Draw.VCPOLYIN.Select(c => c.Item2).ToList();
+                Dictionary<int, Vector3> mappings = new Dictionary<int, Vector3>();
+                foreach (var tup_id_vec in Draw.VCPOLYIN)
+                {
+                    if (!mappings.ContainsKey(tup_id_vec.Item2))
+                    {
+                        mappings.Add(tup_id_vec.Item2, tup_id_vec.Item3);
+                    }
+                }
+                var positions = Draw.VCPOLYIN.Select(c => c.Item3).ToArray();
+                List<int> updatedVertexMapping = Draw.VertexMapping.ToList();
+
+
+                Vector3[] NewVectices = new Vector3[updatedVertexMapping.Count];
+
+                for (var i = 0; i < updatedVertexMapping.Count; i++)
+                {
+                    var v = posBlend[originalVerticesIDS[updatedVertexMapping[i]]];
+                    float x = -1 * v.x; float y = v.y; float z = -1 * v.z;
+                    Vector3 vec3 = new Vector3(x, y, z);
+                    NewVectices[i] = vec3;
+                }
+                adjustedPositons.AddRange(NewVectices);
+            }
+            return adjustedPositons;
```

Comment:
What's going on here exactly ? Is that because indices for the blendshape don't match those of the mesh ?
Is this something that can be pre-computed when importing the blendshape ?